job/job.go

```diff
@@ -90,47 +89,37 @@ func HasActiveWorkflow(ctx context.Context, hwID packet.HardwareID) (bool, error
 }
 
 // CreateFromDHCP looks up hardware using the MAC from cacher to create a job
-func CreateFromDHCP(ctx context.Context, mac net.HardwareAddr, giaddr net.IP, circuitID string) (Job, error) {
+func CreateFromDHCP(ctx context.Context, mac net.HardwareAddr, giaddr net.IP, circuitID string) (context.Context, Job, error) {
 	j := Job{
 		mac:   mac,
 		start: time.Now(),
 	}
 
-	span := trace.SpanFromContext(ctx)
-	span.AddEvent("discoverHardwareFromDHCP",
-		trace.WithAttributes(attribute.String("MAC", mac.String())),
-		trace.WithAttributes(attribute.String("IP", giaddr.String())),
-		trace.WithAttributes(attribute.String("CircuitID", circuitID)),
-	)
-
-	d, err := discoverHardwareFromDHCP(ctx, mac, giaddr, circuitID)
+	ctx, d, err := discoverHardwareFromDHCP(ctx, mac, giaddr, circuitID)
 	if err != nil {
-		span.SetStatus(codes.Error, err.Error())
-
-		return Job{}, errors.WithMessage(err, "discover from dhcp message")
+		return ctx, Job{}, errors.WithMessage(err, "discover from dhcp message")
```

Comment:
ditto, lets return nil here for similar reasons as in fetch.go. Here and in the rest of the function's error branches.

Reply:
> ditto, lets return nil here for similar reasons as in fetch.go. Here and in the rest of the function's error branches.

As I mentioned, I believe this is the wrong move. It's your code so if you really want this change lmk and I'll do it anyways.

Reply:
Yep I see it that in this function the `ctx` is likely different even in the error case (if `j.setup()` is what returns an error) which I hadn't noticed before so sounds fine to not return `nil` as that will be confusing to callers.